prow/sidecar/censor.go

```
@@ -112,6 +116,204 @@ func (o Options) censor() error {
+		if info.IsDir() {
+			return nil
+		}
+
+		wg.Add(1)
+		children.Add(1)
+		go func() {
+			censor(absPath)
+			children.Done()
+		}()
+		return nil
+	}); err != nil {
+		return fmt.Errorf("could not walk unpacked archive to censor them: %w", err)
+	}
+
+	children.Wait()
+	if err := archive(outputDir, archivePath); err != nil {
+		return fmt.Errorf("could not re-pack archive: %w", err)
+	}
+	return nil
+}
+
+// unarchive unpacks the archive into the destination
+func unarchive(archivePath, destPath string) error {
+	input, err := os.Open(archivePath)
+	if err != nil {
+		return fmt.Errorf("could not open archive for unpacking: %w", err)
+	}
+	zipReader, err := gzip.NewReader(input)
+	if err != nil {
+		return fmt.Errorf("could not read archive: %w", err)
+	}
+	tarReader := tar.NewReader(zipReader)
+	defer func() {
+		if err := zipReader.Close(); err != nil {
+			logrus.WithError(err).Warn("Could not close zip reader after unarchiving.")
+		}
+		if err := input.Close(); err != nil {
+			logrus.WithError(err).Warn("Could not close input file after unarchiving.")
+		}
+	}()
+
+	for {
+		entry, err := tarReader.Next()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return fmt.Errorf("could not read archive: %w", err)
+		}
+		if !validRelPath(entry.Name) {
+			return fmt.Errorf("tar contained invalid name error %q", entry.Name)
+		}
+		rel := filepath.FromSlash(entry.Name)
+		abs := filepath.Join(destPath, rel)
+		mode := entry.FileInfo().Mode()
+		switch {
+		case mode.IsDir():
+			if err := os.MkdirAll(abs, 0755); err != nil {
+				return fmt.Errorf("could not create directory while unpacking archive: %w", err)
+			}
+		case mode.IsRegular():
+			file, err := os.OpenFile(abs, os.O_RDWR|os.O_CREATE|os.O_TRUNC, mode.Perm())
+			if err != nil {
+				return err
+			}
+			n, err := io.Copy(file, tarReader)
+			if closeErr := file.Close(); closeErr != nil && err == nil {
+				err = closeErr
```

Lets just use an error slice and utilerrors, potentially swallowing errors seems like it might result in hard to debug issues

---

Good catch, this was copy pasta, can change it